backbone/src/main/java/org/researchstack/backbone/ui/ViewTaskActivity.java

```diff
@@ -369,4 +373,8 @@ public void setActionBarTitle(String title)
             actionBar.setTitle(title);
         }
     }
+
+    public interface ResultListener {
```

Comment:
Nice. This will be a place to hook in that JSON serialization for saving clientData with ScheduleActivity

Reply:
Yeah I think so too.  Just make sure the "instanceof" check and the call to that listener is in a spot where it is called on the previous StepLayout (so it's result is set)